app/components/support/crate-report-form.js

```diff
@@ -0,0 +1,135 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+
+import { dropTask } from 'ember-concurrency';
+import window from 'ember-window-mock';
+import { alias } from 'macro-decorators';
+
+import ajax from '../../utils/ajax';
+
+const REASONS = [
+  {
+    reason: 'spam',
+    description: 'it contains spam',
+  },
+  {
+    reason: 'name-squatting',
+    description: 'it is name-squatting (reserving a crate name without content)',
+  },
+  {
+    reason: 'abuse',
+    description: 'it is abusive or otherwise harmful',
+  },
+  {
+    reason: 'security',
+    description: 'it contains a vulnerability (please try to contact the crate author first)',
+  },
+  {
+    reason: 'other',
+    description: 'it is violating the usage policy in some other way (please specify below)',
+  },
+];
+
+export default class CrateReportForm extends Component {
+  @service store;
+  @alias('loadCrateTask.last.value') crate;
+
+  @tracked name = '';
+  @tracked selectedReasons = [];
+  @tracked detail = '';
+  @tracked nameInvalid;
+  @tracked reasonsInvalid;
+  @tracked detailInvalid;
+
+  reasons = REASONS;
+
+  constructor() {
+    super(...arguments);
+    this.reset();
+    this.name = this.args.name;
+  }
+
+  reset() {
+    this.nameInvalid = false;
+    this.reasonsInvalid = false;
+    this.detailInvalid = false;
+  }
```

Comment:
any reason for not setting them directly in their `@tracked` definitions above? looks like `reset()` is not called anywhere else?

Reply:
Oops, this logic is more relevant to the previous version of the dialog. I'll remove it.